#
# Make this build with a newer erlang by
#    forcing SSL40, which it was using with the old erlang
#    removing LDAP password modify support
#    supply the 64-bit CFLAGS so we don't get ELF mismatches.
# This can be removed when we upgrade to a newer ejabberd.
#
--- ejabberd-2.1.13/src/acinclude.m4.orig	Tue May 19 13:24:30 2015
+++ ejabberd-2.1.13/src/acinclude.m4	Tue May 19 13:25:19 2015
@@ -185,7 +185,7 @@
    # Second line
    ERLANG_EI_LIB=`cat conftest.out | head -n 2 | tail -n 1`
    # Third line
-   ERLANG_SSLVER=`cat conftest.out | head -n 3 | tail -n 1`
+   ERLANG_SSLVER="-DSSL40"
    # End line
    ERLANG_DIR=`cat conftest.out | tail -n 1`
 
--- ejabberd-2.1.13/src/eldap/ELDAPv3.asn.orig	Tue Jun 25 03:46:12 2013
+++ ejabberd-2.1.13/src/eldap/ELDAPv3.asn	Tue May 19 18:22:31 2015
@@ -286,7 +286,7 @@
         responseName     [10] LDAPOID OPTIONAL,
         response         [11] OCTET STRING OPTIONAL }
 
-passwdModifyOID LDAPOID ::= "1.3.6.1.4.1.4203.1.11.1"
+-- passwdModifyOID LDAPOID ::= "1.3.6.1.4.1.4203.1.11.1"
 
 PasswdModifyRequestValue ::= SEQUENCE {
         userIdentity    [0]  OCTET STRING OPTIONAL,

--- ejabberd-2.1.13/src/eldap/eldap.erl	Tue Jun 25 03:46:12 2013
+++ ejabberd-2.1.13/src/eldap/eldap.erl.new	Tue May 19 18:29:14 2015
@@ -734,15 +734,15 @@
 			deleteoldrdn = DelOldRDN,
 			newSuperior  = NewSup}};
 
-gen_req({modify_passwd, DN, Passwd}) ->
-    {ok, ReqVal} = asn1rt:encode(
-		     'ELDAPv3', 'PasswdModifyRequestValue',
-		     #'PasswdModifyRequestValue'{
-				  userIdentity = DN,
-				  newPasswd = Passwd}),
-    {extendedReq,
-     #'ExtendedRequest'{requestName = ?passwdModifyOID,
-			requestValue = list_to_binary(ReqVal)}};
+%% gen_req({modify_passwd, DN, Passwd}) ->
+%%     {ok, ReqVal} = asn1rt:encode(
+%% 		     'ELDAPv3', 'PasswdModifyRequestValue',
+%% 		     #'PasswdModifyRequestValue'{
+%% 				  userIdentity = DN,
+%% 				  newPasswd = Passwd}),
+%%     {extendedReq,
+%%      #'ExtendedRequest'{requestName = ?passwdModifyOID,
+%% 			requestValue = list_to_binary(ReqVal)}};
 
 gen_req({bind, RootDN, Passwd}) ->
     {bindRequest,
